src/Components/Web.JS/src/Services/EventRegistry.ts

```diff
@@ -0,0 +1,66 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+import { IBlazor } from '../GlobalExports';
+
+// The base Blazor event type.
+// Properties listed here get assigned by the event registry in 'dispatchEvent'.
+interface BlazorEvent {
+  blazor: IBlazor;
+  type: keyof BlazorEventMap;
+}
+
+// Maps Blazor event names to the argument type passed to registered listeners.
+export interface BlazorEventMap {
+  'enhancedload': BlazorEvent;
+}
+
+export class EventRegistry {
```

Comment:
Minor point, but this could get confusing since we have `EventDelegator` that also tracks a registry of events (i.e., ones for Blazor components).

Maybe rename to `JSEventRegistry`?